src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -143,6 +176,37 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
             }
         }
 
+        public async Task AddConfiguredMetadataAsync(AzureBlobEgressProviderOptions options, EgressArtifactSettings artifactSettings, IEndpointInfo endpointInfo, CancellationToken token)
+        {
+            DiagnosticsClient client = new DiagnosticsClient(endpointInfo.Endpoint);
+
+            Dictionary<string, string> envBlock = await client.GetProcessEnvironmentAsync(token);
+
+            if (options.IncludeHostNameAsMetadata)
+            {
+                if (envBlock.TryGetValue(HostNameMetadataKey, out string hostNameValue))
```

Comment:
I wasn't sure where we landed on this, so I put this in for now, checking "HOSTNAME" and then "CONTAINERNAME" instead of using `GetHostName`. I haven't pulled out the socket exception logger message yet, but if we decide we want to go this route (or something similar) I can pull that out.